### ISSUE TITLE
Use `addon-test-support/` instead of `test-support/`.

### ISSUE THREAD

**Issue by rwjblue** [ciena-blueplanet/ember-test-utils#40]
The `addon-test-support/` folder provides the following benefits over using `test-support/`:

* Limits the assets to `dist/assets/test-support.js` (instead of `dist/assets/tests.js` which is for app code).
* Easily enforces the namespacing requirements, and reduces extra long naming that exists today

Before:

```js
import {unit} from 'dummy/tests/helpers/ember-test-utils/setup-component-test';
```

After:

```js
import {unit} from 'ember-test-utils/test-support/setup-component-test';
```

**Comment by sglanzer-fire** [comment]
You're a ninja - thanks for the tip 👍

**Comment by job13er** [comment]
@rwjblue I don't see `addon-test-support` referenced anywhere [here](https://ember-cli.com/extending/#addon-project-structure) is that an oversight, or is it only available in a newer version of `ember-cli`? [This issue](https://github.com/ember-cli/ember-cli/issues/4766) seems to indicate it's been available since 2.2 or so, but I can't seem to find any documentation on it anywhere. I just want to make sure that it will work with the versions of `ember-cli` we're using before making the change. Thanks.

**Comment by rwjblue** [comment]
Support was added in https://github.com/ember-cli/ember-cli/pull/5205, which has been around since ember-cli@2.2. Some documentation exists [here](https://ember-cli.com/api/classes/Addon.html#method_treeForAddonTestSupport), but you are absolutely correct that we need to update both guides and API docs with more information.
